### ISSUE TITLE
Get query params from request

### ISSUE THREAD

**Issue by phungleson** [poem-web/poem#113]
Hi Sunli,

I am writing an Endpoint, using `impl Endpoint`, I wonder how can I get query params from request?

I am thinking of using `Extractor` like this, `FromRequest::from_request(&request, &mut body).await?;` I am probably wrong?

Thanks!

**Comment by sunli829** [comment]
Yes, use `Request::split` to get `req` and `body`.

```rust
let (req, body) = req.split();
FromRequest::from_request(&req, &mut body).await?
```

Since the `Query` extractor does not parse anything from `Body`, it can also be like this:

```rust
let (req, body) = req.split();
FromRequest::from_request(&req, &mut Default::default()).await?
```

**Comment by phungleson** [comment]
Thanks, since poem already has helper method like `request.path_param`, wonder if it is a good thing to add `request.params`?

**Comment by sunli829** [comment]
Add helper methods `Request::deserialize_path` and `Request::deserialize_query`.

Released in `v1.0.36` 😁

**Comment by phungleson** [comment]
Thanks! it works well 😄

**Comment by phungleson** [comment]
A bit off-topic and just a suggestion, I used a couple of frameworks and they mostly call something similar to this as `request.params` maybe we can double check with a few prominent ones to see what is the best way to call this.

So that new developers can feel more comfortable with poem when they first pick it up (i.e. reducing mental friction). `deserialize_query` and `deserialize_path` sounds rust-centric and people from Python, JS, or Ruby... might not understand what does it mean.

**Comment by sunli829** [comment]
Released in `v1.0.38`

**Comment by phungleson** [comment]
Awesome, thanks! @sunli829
